app/code/Magento/Directory/Model/Currency/Import/CurrencyConverterApi.php

```diff
@@ -74,11 +74,22 @@ public function fetchRates()
      */
     private function convertBatch($data, $currencyFrom, $currenciesTo)
     {
+        $apiKey = $this->scopeConfig->getValue(
+            'currency/currencyconverterapi/api_key',
+            \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+        );
+        if (empty($apiKey)) {
+            $this->_messages[] = __('No API Key was specified or an invalid API Key was specified.');
```

Comment:
I don't see how it can be not empty here.

Reply:
If user gives an API Key from Currency Setup of backend, $apiKey would be not empty. Otherwise, it would be empty.